pydantic/main.py

```diff
@@ -742,7 +742,7 @@ def __deepcopy__(self: Model, memo: dict[int, Any] | None = None) -> Model:
         # and attempting a deepcopy would be marginally slower.
         _object_setattr(m, '__pydantic_fields_set__', copy(self.__pydantic_fields_set__))
 
-        if self.__pydantic_private__ is None:
+        if not hasattr(self, '__pydantic_private__') or self.__pydantic_private__ is None:
```

Comment:
```suggestion
        if getattr(self, '__pydantic_private__', None) is None:
```

Reply:
I found one existing pattern like this, so replicated it. For reference, the existing pattern using hasattr followed by a normal attribute access was in model_construct.

I think it was done that way for typing, since if I make the change you propose (and also make it in the model_construct for consistency), we get these typing issues:

```
(venv) [dmgrime@dave-laptop:pydantic (9122-deepcopy-pydantic-private)]$ git commit -m "* replace hasattr + obj.xxx with getattr(obj, 'xxx', None)"
don't commit to branch...................................................gPassed
check yaml...........................................(no files to check)Skipped
check toml...........................................(no files to check)Skipped
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
codespell................................................................iPassed
Lint.....................................................................t pushPassed
Usage docs links.........................................................Passed
Typecheck................................................................Failed
- hook id: typecheck
- exit code: 1

/home/dmgrime/pydantic/pydantic/main.py
  /home/dmgrime/pydantic/pydantic/main.py:253:25 - error: Object of type "None" is not subscriptable (reportOptionalSubscript)
  /home/dmgrime/pydantic/pydantic/main.py:730:61 - error: "items" is not a known member of "None" (reportOptionalMemberAccess)
  /home/dmgrime/pydantic/pydantic/main.py:751:70 - error: "items" is not a known member of "None" (reportOptionalMemberAccess)
3 errors, 0 warnings, 0 informations
```

Now, I agree 100% the getattr(x, 'att', None) is better, but I'm not familiar with the type checking system in use with pydantic, or if it would be acceptable to do the (mypy equivalent) of `type: ignore`.  That said, here's the net diff and if there's an easy way to manage the complaints from the type checker, this should be closer?

```
(venv) [dmgrime@dave-laptop:pydantic (9122-deepcopy-pydantic-private)]$ git diff
diff --git a/pydantic/main.py b/pydantic/main.py
index 467fd725..c82a5e2f 100644
--- a/pydantic/main.py
+++ b/pydantic/main.py
@@ -247,7 +247,7 @@ class BaseModel(metaclass=_model_construction.ModelMetaclass):
         if cls.__pydantic_post_init__:
             m.model_post_init(None)
             # update private attributes with values set
-            if hasattr(m, '__pydantic_private__') and m.__pydantic_private__ is not None:
+            if getattr(m, '__pydantic_private__', None) is not None:
                 for k, v in values.items():
                     if k in m.__private_attributes__:
                         m.__pydantic_private__[k] = v
@@ -721,7 +721,7 @@ class BaseModel(metaclass=_model_construction.ModelMetaclass):
         _object_setattr(m, '__pydantic_extra__', copy(self.__pydantic_extra__))
         _object_setattr(m, '__pydantic_fields_set__', copy(self.__pydantic_fields_set__))

-        if not hasattr(self, '__pydantic_private__') or self.__pydantic_private__ is None:
+        if getattr(self, '__pydantic_private__', None) is None:
             _object_setattr(m, '__pydantic_private__', None)
         else:
             _object_setattr(
@@ -742,7 +742,7 @@ class BaseModel(metaclass=_model_construction.ModelMetaclass):
         # and attempting a deepcopy would be marginally slower.
         _object_setattr(m, '__pydantic_fields_set__', copy(self.__pydantic_fields_set__))

-        if not hasattr(self, '__pydantic_private__') or self.__pydantic_private__ is None:
+        if getattr(self, '__pydantic_private__', None) is None:
             _object_setattr(m, '__pydantic_private__', None)
         else:
             _object_setattr(
```